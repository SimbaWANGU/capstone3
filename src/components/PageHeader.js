import React from 'react';
import '../styles/PageHeader.scss';

function PageHeader() {
  return (
    <div className="pageHeader" data-testid="pageHeaderDisplay">
      <h1>Premier League</h1>
      <p>Top Scorers 2021/22</p>
    </div>
  );
}

export default PageHeader;
