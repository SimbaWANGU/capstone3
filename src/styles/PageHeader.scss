$fontColor : rgba(255, 255, 255, 1);
$backgroundColor : rgba(87,136,227, .7);
$backgroundColorGradient: rgba(87,136,227, .5);

.pageHeader {
  background-image: linear-gradient(to left, $backgroundColor, $backgroundColorGradient), url('https://st.depositphotos.com/1004061/4227/i/450/depositphotos_42274531-stock-photo-a-player-shooting-on-goal.jpg');
  background-repeat: no-repeat;
  background-size: cover;
  background-position: center;
  height: 30vh;
  display: flex;
  flex-direction: column;

  h1 {
    margin: auto 10vw 0 auto;
    color: $fontColor;
    font-weight: 500;
  }

  p {
    margin: 0 27.5vw auto auto;
    color: $fontColor;
    font-weight: 500;
  }
}