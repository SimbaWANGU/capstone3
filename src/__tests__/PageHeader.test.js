import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import PageHeader from '../components/PageHeader';

describe('Home page header section tests', () => {
  it('Component renders as expected', () => {
    const component = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <PageHeader />
        </BrowserRouter>
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Component is visible to the user', () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <BrowserRouter>
          <PageHeader />
        </BrowserRouter>
      </Provider>,
    );
    expect(getByTestId('pageHeaderDisplay')).toBeInTheDocument();
  });
});
