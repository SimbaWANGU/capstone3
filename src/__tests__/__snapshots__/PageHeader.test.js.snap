// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page header section tests Component renders as expected 1`] = `
<div
  className="pageHeader"
  data-testid="pageHeaderDisplay"
>
  <h1>
    Premier League
  </h1>
  <p>
    Top Scorers 2021/22
  </p>
</div>
`;
